### DISCUSSION TITLE
Is there any way to add custom constraint without losing(or re-implement by copy-and-paste) vrp-pragmatic features?

### DISCUSSION THREAD

**Discussion by pocket7878** [reinterpretcat/vrp#151] [Q&A] [answered]
Sometimes I want to add custom constraint to solver. I can implement it by Feature trait.
But how can I use it without losing other existing features on vrp-pragmatic (e.g skill constraint)?
It looks like my Feature should be added to GoalContext, but GoalContext doesn't have an interface to add feature after creation.
So I'm always copy-and-pasting codes from vrp-pragmatic, or just give-up and use vrp-core directly without vrp-pragmatic features.
Please let me know if there is a better way to do it!

**Comment by reinterpretcat** [comment, marked as answer]
Hi, I would say, until now, the focus was mostly on CLI usage, not as library. I have to admit this is a miss from my side. I will try to prioritize the extensibility of crates as my next task (it seems as interesting architectural exercise).
 I was thinking also to expose python interface for extension, but seems this will require quite of work even with help of PYO3.
